src/main/java/org/opentripplanner/prediction_client/VehiclePresenceGetter.java

```diff
@@ -0,0 +1,56 @@
+package org.opentripplanner.prediction_client;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.message.BasicNameValuePair;
+import org.locationtech.jts.geom.Envelope;
+import org.opentripplanner.hasura_client.HasuraGetter;
+import org.opentripplanner.hasura_client.mappers.HasuraToOTPMapper;
+import org.opentripplanner.profile.Option;
+import org.opentripplanner.util.HttpUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toList;
+
+public class VehiclePresenceGetter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(HasuraGetter.class);
+
+    public Optional<VehiclePresence> getPrediction(String url, Map<String, String> params) {
+        URI uri;
+        try {
+            uri = new URIBuilder(url).addParameters(
+                    params.entrySet().stream().map(it -> new BasicNameValuePair(it.getKey(), it.getValue())).collect(toList())
+            ).build();
+        } catch (URISyntaxException e) {
+            LOG.error("Cannot construct uri for fetching vehicle presence prediction heatmap");
+            return Optional.empty();
+        }
+        VehiclePresence response = HttpUtils.getData(uri, new TypeReference<>() {});
+        LOG.info("Got {} objects from API", response != null ? "1" : "null");
+        return Optional.ofNullable(response).map(this::mapVehiclePresence);
+    }
+
+    VehiclePresence mapVehiclePresence(VehiclePresence vehiclePresence) {
```

Comment:
This function is copied from `VehiclePresenceMapper`

Reply:
I will remove Mapper class as it is redundant - artefact from times when I tired to conform to `HasuraGetter` interface